docs/rules/typescript/prefer-pascal-case-enums.md

```diff
@@ -0,0 +1,33 @@
+# Enforce pascal case when naming enums. (prefer-pascal-case-enums)
```

Comment:
This should be a file rename.  I still see `docs/rules/prefer-pascal-case-enums.md`.

Reply:
I like fully namespacing the rule name because I tend to copy+paste the names from here into `package.json#eslintConfig`.

So `(shopify/typescript/prefer-pascal-case-enums)` would be nice.

Reply:
Nit: Pascal is a noun, so `pascal case` => `Pascal case` 🤓